src/views/_shared/GoogleOAuth.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from "react";
+import GoogleLogin from "react-google-login";
+// import { LOGIN } from "../../graphql-requests/mutations";
+//import { createBrowserHistory } from 'history';
+import { withRouter } from "react-router-dom";
+
+//let history = createBrowserHistory();
+
+class GoogleOAuth extends Component {
+   constructor(props) {
+      super(props);
+      this.state = {};
+   }
+   signup(res) {
+      const googleresponse = {
+         username: res.profileObj.name,
+         first_name: res.profileObj.name,
+         last_name: res.profileObj.familyName,
+         email: res.profileObj.email,
+         token: res.googleId,
+         Image: res.profileObj.imageUrl,
+         ProviderId: "Google",
+      };
+      //debugger;
+   }
+
+   render() {
+      const responseGoogle = response => {
+         console.log(response);
+         let res = response.profileObj;
+         console.log(res);
+         localStorage.setItem("username", res.familyName);
+         localStorage.setItem("first_name", res.familyName);
+         localStorage.setItem("last_name", res.familyName);
+         localStorage.setItem("email", res.email);
+         localStorage.setItem("token", response.googleId);
+         localStorage.setItem("user_id", response.googleId);
+         this.props.history.push("/homeg");
+         //debugger;
+         alert(
+            "Please enter your profile info on the dashboard for future access to your profile!"
+         );
+         this.signup(response);
+      };
+
+      return (
+         <div className="App">
+            <div className="row">
+               <div className="col-sm-12 btn btn-info"></div>
+            </div>
+            <div className="row">
+               <div style={{ paddingTop: "20px" }} className="col-sm-12">
+                  <div className="col-sm-4"></div>
+                  <div className="col-sm-4">
+                     <GoogleLogin
+                        clientId="403619985302-kcu2rduts2gj7oolnvjd89aj5lhgnkf5.apps.googleusercontent.com"
```

Comment:
I think we should abstract away any sort of hard-coded ID into an environment variable. This would help security and also in the event we need to update our credentials. It would allow us to change things in only one place instead of requiring us to hunt down a large ID string if we make changes.

Reply:
Google suggests it be done this way https://developers.google.com/identity/sign-in/web/server-side-flow remeber the secret variable is not accessible and no one can log in to the app info hosted on google without the secret and credentials, I don't how this can be put in production with this current structure